src/main/java/com/endava/endabank/service/impl/ForgotUserPasswordTokenServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.endava.endabank.service.impl;
+
+import com.endava.endabank.constants.Strings;
+import com.endava.endabank.dao.ForgotUserPasswordTokenDao;
+import com.endava.endabank.model.ForgotUserPasswordToken;
+import com.endava.endabank.service.ForgotUserPasswordTokenService;
+import lombok.AllArgsConstructor;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+
+@Service
+@AllArgsConstructor
+public class ForgotUserPasswordTokenServiceImpl implements ForgotUserPasswordTokenService {
+    private ForgotUserPasswordTokenDao forgotUserPasswordTokenDao;
+
+
+    @Override
+    public ForgotUserPasswordToken findByUserId(Integer id) {
+        return forgotUserPasswordTokenDao.findByUserId(id).
+                orElseThrow(() -> new UsernameNotFoundException(Strings.USER_TOKEN_NOT_FOUND));
+    }
+
+    @Override
+    public void save(ForgotUserPasswordToken forgotUserPasswordTokenNew) {
+        Optional<ForgotUserPasswordToken> forgotUserPasswordTokenDbOpt =
+                forgotUserPasswordTokenDao.findByUserId(forgotUserPasswordTokenNew.getUser().getId());
+        if (forgotUserPasswordTokenDbOpt.isPresent()) {
+            ForgotUserPasswordToken forgotUserPasswordTokenDb = forgotUserPasswordTokenDbOpt.get();
```

Comment:
Hi,

You could replace the lines 31,32, 33 for 
`forgotUserPasswordTokenNew = forgotUserPasswordTokenDbOpt.get();
            forgotUserPasswordTokenNew.setToken(forgotUserPasswordTokenNew.getToken());`